### PR TITLE
Update/ci image

### PR DESCRIPTION
**Pull request by scott062**

### CODE REVIEW

**Comment by elizabeth-jimenez** [comment]
Test

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: 2.1
 
 default-python-node: &default-python-node
-  image: circleci/python:3.6.8
+  image: cimg/python:3.6.14
```
Comment (elizabeth-jimenez) [inline .circleci/config.yml:4]: 👀

**Comment by scott062** [comment]
Changed my mind - going to try and get all the warnings resolved in this pr unless there is any new API prs waiting

**Comment by M00staff** [comment]
are the warnings because of the bump in version - or do i need to tinker with my IDE's python settings?

**Comment by scott062** [comment]
Ideally everyone should be following pep8 standard and the pylint should enforce that. Obviously within reason so we can add some ignores as needed. The linter stuff has been there the whole time but the tests couldn't run so we didn't pay close attention to the other jobs running in our pipeline. This is resolving both issues - the test suite should run and we should be close to zero warnings/errors on linter side too

File: talentmap_api/common/common_helpers.py [hunk 2]
```diff
@@ -503,7 +503,7 @@ def formatCSV(data, fieldsInfo):
 
     # setting up the defaults
     for x in fields_formatted:
-        if fields_formatted[x] is "" or fields_formatted[x] is None:
+        if fields_formatted[x] == "" or fields_formatted[x] is None:
```
Comment (scott062) [inline talentmap_api/common/common_helpers.py:506]: @elizabeth-jimenez good example of what we were talking about

File: talentmap_api/common/common_helpers.py [hunk 3]
```diff
@@ -576,31 +576,31 @@ def registered_handshake_notification_thread(cp_id, jwt, perdet_to_exclude, is_a
...
...AsRkivBvVQgrJPmfhBGZQKcpHPWxTuI14NCDXgV8YS8tuKtzHquQt0bqxKV3uWepPnn3t0fe+8+c2rwpUmjJq86qGbRiuyf0Z6u8KT9L/++Qlvak1KOfWPGpLHKL9eyHX+WcWnqDvdxhYV0KcanLL13/P3hGm7oLtdXF9+3Xf79fdAKkNVLxjAT5bpoDjvtIa/X4FTJUQRIEiMmF4iIe3zhYfHqGEyr165zRtItCeLgMbTCZUcxhonbJ0cDI9tyLzC6MbAJtWJ23LwklORFdJAHAsFGhogWlr/99tvuf0KuLJk5KD7wcJDUjMG7bmOHSzy3sCsMYGbUzVMShpyRV37f4R0zaF+FJwxsvB+EGcGHWztceV3ExdsbUzkLGOdjh+zrEo5HADFywbE9M0Z5fS/f4D90Z+azeM+QUce75OqFjyx0x3LYITXu/ccM7uFyT/iML3y6pzP6Meibzt3oDT/5q17PYX/tbf3tBG/Z0sNcmJdtj8/G80CI1M/TeRu3+tsfN26s+xvR9sHqNzPeip07a70NH9Z3EhO/+OMuJzpOO6LOG5Led/p/ILIAbw/bxytix0Yy+hBfuATH7O2P9nj/5YQad36CwsNAHBJ2ZefGKpUhRBAcN1zZ+bymwun2ul4z5Q7D4tonlyNJ5SvCqa65PFWVLujlQIhwL7GtYqtg2b21adW9/iOV31Tbs6HtyPN/fbS+WoUQQghRNgHS0dHR8O4v8YJsaujti4+hvggxQ6cYAwfD5ts3TsuIB1f1Z958F78+yjegfr/sYe+EiVfl7J7Oe19Y/EDOvJH9hM/FF3sv71rtfXDO4EwSOrkeeE2CCetBAYKwIeH8+ku3dDJS8W7M/8XejPE65b/vM3D5n5n28Ex8GAzd51/zvOde7XB/W/NBq/RE8jVegg+3pv7/4qd6Zt6DOGHWHwFDHkQUCIlf/GGXN/CI4503x6punTZkq3sN4UHOBbBdtg8vv7vH+/TIPd7Mr6zzeg840vv1c5u8OQv7ZcQEr+PB2JFqFu/EDCJh56BPuypkJgj5LHp2sN2wR8cgqR4BEfRqcPyIE7bPfnH8wc8iR4b3ML54L8462YsVesB5YXxfXel5zb/YmxGIi/+phxt7PFhXnlfrTftCZ3H5g0f7ueMpZxJ6PqKeXI6HFj7p3s89gPcwmFROjhT3U6FNDM2LAu/+8tRg9asZo
```
Comment (scott062) [inline talentmap_api/common/common_helpers.py:603]: Just added disable line length for these long strings

File: talentmap_api/common/tests/test_common_helpers.py [hunk 4]
```diff
@@ -1,19 +1,16 @@
...
 from model_mommy import mommy
 
-from talentmap_api.common.common_helpers import get_permission_by_name, get_group_by_name, in_group_or_403, has_permission_or_403, ensure_date, safe_navigation, order_dict, serialize_instance, in_superuser_group, validate_filters_exist
+from talentmap_api.common.common_helpers import get_group_by_name, in_group_or_403, ensure_date, order_dict, in_superuser_group
 
 @pytest.mark.django_db()
 def test_ensure_date():
 
-    ensure_date(201225123) == None
+    ensure_date(201225123) is None
```
Comment (scott062) [inline talentmap_api/common/tests/test_common_helpers.py:13]: @elizabeth-jimenez Same as above

File: talentmap_api/common/tests/test_sharing.py [hunk 5]
```diff
@@ -1,7 +0,0 @@
-import pytest
-import json
-
-from django.contrib.auth.models import User
-
```
Comment (scott062) [inline talentmap_api/common/tests/test_sharing.py:5]: No idea what this super old empty file is for so removed it. May have been a shared file for all tests but is no longer used

File: talentmap_api/fsbid/services/agenda_item_validator.py [hunk 6]
```diff
@@ -91,7 +78,7 @@ def validate_legs(legs):
     }
 
     # AI Legs - must not be empty
-    if not len(legs):
+    if legs:
```
Comment (scott062) [inline talentmap_api/fsbid/services/agenda_item_validator.py:81]: @elizabeth-jimenez theres a specific `len` error for this - basically just to use the implied truthiness of lists with content vs the falsey value of empty lists.

File: talentmap_api/fsbid/services/assignment_history.py [hunk 7]
```diff
@@ -167,7 +167,6 @@ def fsbid_assignments_to_talentmap_assignments(data):
         "asgs_update_id": "asgsupdateid",
         "asgs_update_date": "asgsupdatedate",
         "position": "position",
-        "asgd_tod_code": "asgdtodcode",
```
Comment (scott062) [inline talentmap_api/fsbid/services/assignment_history.py:170]: mapped twice

File: talentmap_api/fsbid/services/bid.py [hunk 8]
```diff
@@ -114,16 +117,16 @@ def map_bids_to_disable_handshake_if_accepted(bids):
...
-                and not pydash.find(hasAcceptedHandshakeIds, lambda y: y['cp_id'] == pydash.get(x, 'position_info.id'))
-                else False,
-        })
+        **x,
+        'accept_handshake_disabled': True if pydash.find(hasAcceptedHandshakeIds, lambda y:
+                                                             y['cycle_id'] == pydash.get(x, 'position_info.bidcycle.id'))
+                                       and not pydash.find(hasAcceptedHandshakeIds, lambda y: y['cp_id'] == pydash.get(x, 'position_info.id'))
+                                       else False,
+    })
     return bidsClone
```
Comment (scott062) [inline talentmap_api/fsbid/services/bid.py:130]: Just what the linter wanted but not sure if it's better or worse

**Comment by scott062** [comment]
image
Some warnings for unused imports especially for coreapi in our views - still need to test without it. Only concern is that we don't use it ANYWHERE in the project so may be a missing dependency needed for our other imports like the swagger docs

File: talentmap_api/fsbid/services/agenda_item_validator.py (hunk 6 above)
Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/services/agenda_item_validator.py:81]: Nice nice

File: talentmap_api/fsbid/services/bid.py [hunk 9]
```diff
@@ -36,7 +36,10 @@ def user_bids(employee_id, jwt_token, position_id=None, query={}):
     filteredBids = {}
     # Filter out any bids with a status of "D" (deleted)
     filteredBids['Data'] = [b for b in list(pydash.get(bids, 'Data') or []) if smart_str(b["bs_cd"]) != 'D']
-    mappedBids = [fsbid_bid_to_talentmap_bid(bid, jwt_token) for bid in filteredBids.get('Data', []) if bid.get('cp_id') == int(position_id)] if position_id else map(lambda b: fsbid_bid_to_talentmap_bid(b, jwt_token), filteredBids.get('Data', []))
```
[the remark this answers is not in the document]
Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/services/bid.py:39]: 😂 ❤️

File: talentmap_api/stats/views.py [hunk 10]
```diff
@@ -149,4 +148,10 @@ class ViewPositionDistinctListView(mixins.ListModelMixin,
     permission_classes = (IsAuthenticated, isDjangoGroupMember('superuser'))
 
     def get_queryset(self):
-        return get_prefetched_filtered_queryset(ViewPositionInstance, self.serializer_class).annotate(distinct_name=Concat('position_type', 'position_id', 'user_id', 'date_of_view_week', output_field=TextField())).order_by('distinct_name').values('distinct_name').distinct()
+        return (
+            get_prefetched_filtered_queryset(ViewPositionInstance, self.serializer_class)
```
Comment (elizabeth-jimenez) [inline talentmap_api/stats/views.py:152]: 🏅

File: talentmap_api/common/common_helpers.py (hunk 2 above)
Comment (elizabeth-jimenez) [reply talentmap_api/common/common_helpers.py:506]: ⭐

File: talentmap_api/fsbid/services/agenda_item_validator.py [hunk 11]
```diff
@@ -1,19 +1,6 @@
...
-from urllib.parse import urlencode, quote
-from functools import partial
-from copy import deepcopy
-
-from django.utils.encoding import smart_str
-from django.conf import settings
-from django.http import HttpResponse
 
-from talentmap_api.fsbid.services import common as services
+from django.conf import settings
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/agenda_item_validator.py:3]: What did you say the order/groupings of these should be?

File: talentmap_api/organization/views.py [hunk 12]
```diff
@@ -1,6 +1,6 @@
-from talentmap_api.common.cache.views import CachedViewSet
+from talentmap_api.common.cache.views import CachedViewSet # pylint: disable=unused-import
 
-from talentmap_api.common.common_helpers import get_prefetched_filtered_queryset
-from talentmap_api.common.mixins import FieldLimitableSerializerMixin
+from talentmap_api.common.common_helpers import get_prefetched_filtered_queryset # pylint: disable=unused-import
+from talentmap_api.common.mixins import FieldLimitableSerializerMixin # pylint: disable=unused-import
```
Comment (elizabeth-jimenez) [inline talentmap_api/organization/views.py:4]: u scared to delete? 😈  🐈 👻

File: talentmap_api/organization/filters.py [hunk 13]
```diff
@@ -1,7 +1,7 @@
-import rest_framework_filters as filters
+import rest_framework_filters as filters # pylint: disable=unused-import
 
-from talentmap_api.organization.models import Obc
-from talentmap_api.common.filters import multi_field_filter, negate_boolean_filter, full_text_search, NumberInFilter
-from talentmap_api.common.filters import ALL_TEXT_LOOKUPS, INTEGER_LOOKUPS, FOREIGN_KEY_LOOKUPS
+from talentmap_api.organization.models import Obc # pylint: disable=unused-import
+from talentmap_api.common.filters import multi_field_filter, negate_boolean_filter, full_text_search, NumberInFilter # pylint: disable=unused-import
+from talentmap_api.common.filters import ALL_TEXT_LOOKUPS, INTEGER_LOOKUPS, FOREIGN_KEY_LOOKUPS # pylint: disable=unused-import
```
Comment (elizabeth-jimenez) [inline talentmap_api/organization/filters.py:5]: 🙈

File: talentmap_api/fsbid/services/agenda_item_validator.py (hunk 11 above)
Comment (scott062) [reply talentmap_api/fsbid/services/agenda_item_validator.py:3]: Standard, 3rd Party, Project

File: talentmap_api/organization/views.py (hunk 12 above)
Comment (scott062) [reply talentmap_api/organization/views.py:4]: yeah - idk if we need these files for obc stuff in upcoming pages so i just decided to wait

File: talentmap_api/fsbid/services/bureau.py [hunk 14]
```diff
@@ -177,11 +171,7 @@ def fsbid_bureau_position_bids_to_talentmap(bid, jwt, cp_id, active_perdet):
 
     active_handshake_perdet = None
     if active_perdet:
-        if int(active_perdet) == int(emp_id):
-            active_handshake_perdet = True
-        else:
-            active_handshake_perdet = False
-
+        active_handshake_perdet = int(active_perdet) == int(emp_id)
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/bureau.py:174]: 🌟

File: talentmap_api/fsbid/services/bureau.py [hunk 15]
```diff
@@ -177,11 +171,7 @@ def fsbid_bureau_position_bids_to_talentmap(bid, jwt, cp_id, active_perdet):
...
     fullname = bid.get("full_name", None)
     if fullname:
```
Comment (scott062) [inline talentmap_api/fsbid/services/bureau.py:176]: another risk - heads up @elizabeth-jimenez

File: talentmap_api/fsbid/services/agenda_item_validator.py (hunk 11 above)
Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/services/agenda_item_validator.py:3]: Thanks!

**Comment by elizabeth-jimenez** [comment]
@scott062 finished looking through all the files - will user test PR as I review the FE PRs ready for review 🎊

**Review by M00staff — APPROVED** [review]
I think Elizabeth will be OoO - I've been on this branch all week, working fine for me

**Pull request merged**